packages/commerce-sdk-react/src/hooks/useAuthenticatedRequest.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2022, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import {QueryFunctionContext, QueryKey} from '@tanstack/react-query'
+import useAuth from './useAuth'
+import useCommerceApi from './useCommerceApi'
+import {ApiClients, QueryFunction, MutationFunction} from './types'
+
+interface Client {
+    clientConfig: {
+        headers: Record<string, string>
+    }
+}
+
+function useAuthenticatedMutation<TData, TVariables = unknown>(
+    fn: MutationFunction<TData, TVariables>
+) {
+    const auth = useAuth()
+    const apiClients = (useCommerceApi() as unknown) as Record<string, Client>
+
+    return (variables: TVariables) => {
+        return auth
+            .ready()
+            .then(({access_token}) => {
```

Comment:
These two functions have the same function signature and implementation. Can we do something like this:
```
function useAuthenticated<TData, TVariables = unknown>(
    fn: MutationFunction<TData, TVariables>
)
function useAuthenticated<TData, TVariables = unknown>(
    fn: QueryFunction<TData, TQueryKey>
)
function useAuthenticated<TData, TVariables = unknown>(
    fn: QueryFunction<TData, TQueryKey> | MutationFunction<TData, TVariables>
) {

 return (returnedVal: TVariables | QueryFunctionContext<TQueryKey>) => {
    return auth.ready() ....
  }
}
```